### PR TITLE
dev/core#2721 [REF] Further divide savedSearchParam loading into the sql functions

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
dev/core#2721 [REF] Further divide savedSearchParam loading into the sql functions

Before
----------------------------------------
handling for getting the savedSearch params from the saved search all munged up

After
----------------------------------------
split out to the type of search they relate to

Technical Details
----------------------------------------
Builds on #20953

Comments
----------------------------------------

### CODE REVIEW

**Comment by colemanw** [comment]
Test failures look related

**Comment by eileenmcnaughton** [comment]
yep I messed up a variable - but noting that there is test cover in these functions

CRM_Contact_Form_Search_Custom_GroupTest.testCount with data set #15
CRM_Contact_Form_Search_Custom_GroupTest.testCount with data set #16
CRM_Contact_Form_Search_Custom_GroupTest.testCount with data set #17
CRM_Contact_Form_Search_Custom_GroupTest.testAll with data set #15
CRM_Contact_Form_Search_Custom_GroupTest.testAll with data set #16
CRM_Contact_Form_Search_Custom_GroupTest.testAll with data set #17
CRM_Contact_Form_Search_Custom_GroupTest.testContactIDs with data set #15
CRM_Contact_Form_Search_Custom_GroupTest.testContactIDs with data set #16
CRM_Contact_Form_Search_Custom_GroupTest.testContactIDs with data set #17
api.v4.Entity.SavedSearchTest.testContactSmartGroup
api.v4.Entity.SavedSearchTest.testEmailSmartGroup
api.v4.Entity.SavedSearchTest.testSmartGroupWithHaving
api.v4.Entity.SavedSearchTest.testMultipleSmartGroups

File: CRM/Contact/BAO/GroupContactCache.php [hunk 1]
```diff
@@ -796,27 +815,27 @@ private static function updateCacheFromTempTable(CRM_Utils_SQL_TempTable $groupC
...
+        $sql = self::getApiSQL($savedSearch, $addSelect, $excludeClause);
       }
       else {
-        // CRM-7021 rectify params to what proximity search expects if there is a value for prox_distance
-        if (!empty($ssParams)) {
-          CRM_Contact_BAO_ProximityQuery::fixInputParams($ssParams);
-        }
-        if (isset($ssParams['customSearchID'])) {
-          $sql = self::getCustomSearchSQL($savedSearchID, $ssParams, $addSelect, $excludeClause);
+        if (!empty($savedSearch['form_values']['customSearchID'])) {
```
Comment (seamuslee001) [inline CRM/Contact/BAO/GroupContactCache.php:834]: @eileenmcnaughton is this right why is it buried in form_values?

Comment (eileenmcnaughton) [reply CRM/Contact/BAO/GroupContactCache.php:834]: Yeah - why is it! But, it is. that form_values field saved to the db is what holds it

**Comment by seamuslee001** [comment]
ok am comfortable with the unit tests covering us here and reviewed the extraction and seems to make sense merging

**Pull request merged**